vm/src/vm/mod.rs

```diff
@@ -763,6 +768,22 @@ impl VirtualMachine {
                 self.print_exception(exc);
                 #[cfg(unix)]
                 {
+                    unsafe {
+                        if sigaction(
+                            SIGINT,
+                            &SigAction::new(
+                                nix::sys::signal::SigHandler::SigDfl,
+                                SaFlags::SA_ONSTACK,
+                                SigSet::empty(),
+                            ),
+                        )
+                        .is_ok()
+                        {
+                            interpreter::flush_std(self);
+                            kill(getpid(), SIGINT).expect("Expect to be killed.");
+                        }
+                    }
+
```

Comment:
This seems like a lot of `unsafe`.

---

Does Rust allow, e.g.
```rust
if (unsafe { dangerous() }) {
    // ...
}
```

…or is the syntax too restrictive for that?

Reply:
I didn't know the syntax is available. Only `sigaction` function needs `unsafe`. So I'll apply it like with `cargo fmt`.

```suggestion
                        if unsafe { sigaction(
                            SIGINT,
                            &SigAction::new(
                                nix::sys::signal::SigHandler::SigDfl,
                                SaFlags::SA_ONSTACK,
                                SigSet::empty(),
                            ),
                        ) }
                        .is_ok()
                        {
                            interpreter::flush_std(self);
                            kill(getpid(), SIGINT).expect("Expect to be killed.");
                        }

```

Reply:
in general, you may want to make a new variable for complex expressions.
```
let r = unsafe { sigaction(
                            SIGINT,
                            &SigAction::new(
                                nix::sys::signal::SigHandler::SigDfl,
                                SaFlags::SA_ONSTACK,
                                SigSet::empty(),
                            ),
                        ) };
if r.is_ok() {
    ...
}
```

Reply:
I applied it in [a3507e7](https://github.com/RustPython/RustPython/pull/4454/commits/a3507e7f58aaa13897cb5d15608ab4499f50f415) commit 🙏🏻